hoomd/hpmc/integrate.py

```diff
@@ -1460,6 +1463,32 @@ class SphereUnion(_HPMCIntegrator):
                                       (0.25, 0.0, 0.0)]);
         mc.shape["B"] = dict(diameters=[0.05], centers=[(0.0, 0.0, 0.0)]);
         mc.depletant_fugacity["B"] = 3.0
+
+    Attributes:
+        shape (TypeParameter[particle type, dict]):
+            The shape parameters for each particle type. The dictionary has the
+            following keys:
+
+            * ``shapes`` (List[dict], **required**) -
+              Shape parameters for each sphere in the union. See `Sphere.shape`
+              for the accepted parameters.
+            * ``positions`` (List[Tuple[float, float, float]], **required**) -
+              Position of each sphere in the union.
+            * ``orientations`` (List[Tuple[float, float, float, float]],\
+                                **default:** None) -
+              Orientation of each sphere in the union. When not None,
+              ``orientations`` must have a length equal to that of
+              ``positions``. When None (the default), ``orientations`` is
+              initialized with all [1,0,0,0]'s.
+            * ``overlap`` (List[bool], **default:** None) - Check for overlaps
+              between constituent particles when ``overlap [i] & overlap[j]``
+              is nonzero (``&`` is the bitwise AND operator). When not None,
+              ``overlap`` must have a length equal to that of ``faces``. When
+              None (the default), ``overlap`` is initialized with all 1's.
```

Comment:
While true on the C++ side, users should only see Boolean values, so should this be all `True`s?

Reply:
I thought I saw a commit where @jglaser changed it from bool back to int. I'm assuming that there is some reason for this (e.g. to treat these as bit fields for groups). Is that true @jglaser ? Or should `overlap` be boolean?

Reply:
This is a bit mask. Perhaps we should rename it to `mask` to make this more explicit.

Reply:
what is the bit mask for?

Reply:
If it is user facing, we need to document its use and keep it as an int with a name like `overlap_mask`.

If not, does setting this to all ones by default break anything? If not we can still pretend it is a `bool` on the Python side, and know that C++ will overwrite the 1's when necessary.

Reply:
This bit mask allows the user to select certain constituent particles for overlap checking. **Example**: one union with two constituent particle types (read, patches) with `overlap=1` and `overlap=3`. Now have a second union particle (read, depletant), a single sphere with `overlap=2`. It will only interact with the second type of patch, but not with the first one. Nevertheless, no patch can cross any other. `1` is a reasonable default. It should be `int` not `bool`.

Reply:
Okay, I see we document this by describing the bit wise AND. Thanks for the explanation @jglaser.